Db fake

DiegoBrian opened the pull request:
Branch de inserção de dados no BDF
Resolve Issue #13

### BDF
Se caracteriza por um banco de dados _fake_ em memória formado por listas contendo entidades de uma única natureza. Exemplo: BDF = {  List< Oportunidade >, List< Tags >, ... }.

Sua classe `br.unb.unbsolidaria.persistencia.BancoDeDados` utiliza o padrão de projeto [Singleton](https://en.wikipedia.org/wiki/Singleton_pattern), o que permite que somente uma única instância deste BancoDeDados fake forneça as informações para todos as outras classes por meio do método `BancoDeDados.getInstance()`.

No momento não há componentes do projeto que utilizam esta classe. Portanto, efeitos colaterais no app seriam nulo.

DiegoBrian committed:
Insercao de dados BD Fake

DiegoBrian committed:
Alterar forma de aramazenamento da data de nascimento

DiegoBrian committed:
Adição de Voluntários

DiegoBrian committed:
Adição de Organizações

DiegoBrian committed:
Atualização das descrições

DiegoBrian committed:
Adição de Oportunidades

DiegoBrian committed:
Atualização das descrições

asm95 commented:
Prós:
- Resolve Issue #13

Contra:
- Classe não compila, pois não foi testado e as instanciações das entidades não possuem parâmetros de acordo com a especificação das entidades. Exemplo:

`new Tags(0, "Marcenaria", "Trabalhos com madeira")` - Erro: o primeiro parâmetro recebe uma String e não um Integer.

Ação:
- Corrigir erros de compilação. 
  Status: @asm95 está corrigindo e um commit será realizado em breve.

## 

**Edit 1**
- Agora o pull-request pode ser _aceito_ utilizando a modalidade "Squash and merge", visto que é útil quando temos em mente descartar uma branch[1]. Isto é, como a Issue #13 foi resolvida, a branch BDFake não é mais necessária.
  Status: @asm95 irá simular o merge na branch master a fim de verificar o funcionamento

**Edit 2**
- Nenhum conflito: Gradle build realizado com sucesso. Pull request pronto para ser aceito.

### Referências

[1] http://stackoverflow.com/a/2427520

asm95 committed:
Viabilização do pull-request #20

- Corrige erros de compilação da classe BancoDeDados
- Adicionada classe de Teste do Banco de Dados

Obs: para fazer o BancoDeDados _fake_ funcionar foi necessário realizar
as seguintes alterações:

- Mudança do tipo mId em _Tags_ para int. Razão: HashMap pode ser feito
em cima do atributo mNome, pois é do tipo String.

- Mudança do tipo mDataNascimento para java.util.Calendar. Isto porque a
framework android.icu.util.Calendar é suportada somente a partir da API
24 (Android 7.0). Isto compromete a _compatibilidade_ com versões
anteriores do Android. A não ser que o APK possua o ICU4J também.
Afetam Classes: Oportunidade, Voluntário

- Corrigido nome do método getmOrgs para getOrgs em _Oportunidade_
Afetam Classes: Oportunidade
